src/renderer/screens/KeyboardSelect.js

```diff
@@ -148,8 +148,15 @@ class KeyboardSelect extends React.Component {
     return new Promise(resolve => {
       focus
         .find(...Hardware.serial)
-        .then(devices => {
-          const list = this.findNonSerialKeyboards(devices);
+        .then(async devices => {
+          let newDevices;
+          await focus.open(devices[0].comName, devices[0]);
+          let keyboardType = await focus.getKeyboardType();
+          focus.close();
+          newDevices = devices.filter(
+            item => keyboardType.trim() === item.device.info.keyboardType
```

Comment:
If I understand this correctly, if the device descriptor does not have a `keyboardType` property, and the device itself doesn't implement the focus command either, then this filter should succeed, and let the keyboard through, right?

The reason I'm asking is because I think that the focus command should be optional, for devices where the physical layout is fixed (to save space, mostly).

Reply:
Thank you for the question.
Yes. So now it should works good.

Resolved [here](https://github.com/keyboardio/Chrysalis/pull/400/commits/0479b1e9dfeb17b297ea5293525aa05194a4e501)